### PR TITLE
NEO-2280 // Go To functionaility

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to Table story](https://deploy-preview-511--neo-react-library-storybook.netlify.app/?path=/story/components-table--server-side-pagination)
[Link to Figma reference](https://www.figma.com/design/SQHQlIhFHWCjngHZOnK8MH/Web-Components-2.0?node-id=6170-2353&t=gPh9ofqp708VwfSe-0)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

Adding basic "go to" functionality to pagination. 

**IMPORTANT**: I tested with VO and a11y is definitely not working as intended. I've [added a note](https://teams.microsoft.com/l/message/19:51f2a46da4674b958166ab02f482a977@thread.tacv2/1724875660320?tenantId=04a2636c-326d-48ff-93f8-709875bd3aa9&groupId=853bb606-e586-482d-9bb8-a0a0af53510d&parentMessageId=1724859305794&teamName=UX%2FUI%20Team%20%F0%9F%8E%A8&channelName=Accessibility%20Q%20and%20A&createdTime=1724875660320) to our next Terri session to ask her what is supposed to happen. 

`@avaya-dux/dux-design`: Review figma and implementation for matching visuals.

- confirm that the visuals match (or are close enough)
- confirm that the "go to" input works as expected
- if there are improvements to be made, note them so we can discuss if they should be done in this PR or a future one
- note that the input width changes based on maximum page number, please update the row count to see different width
 - for example, use [this story](https://deploy-preview-511--neo-react-library-storybook.netlify.app/?path=/story/components-table--default) to see smaller input width, and [this story](https://deploy-preview-511--neo-react-library-storybook.netlify.app/?path=/story/components-table--server-side-pagination) for large input wdith

`@avaya-dux/dux-devs`: General functionality was approved in [previous PR](https://github.com/avaya-dux/neo-react-library/pull/503). Double check for bugs.

## Summary by CodeRabbit

- **New Features**
  - Enhanced input handling in the pagination component for improved robustness when entering page numbers.
  - Improved rendering of pagination text for better clarity and structure.

- **Style**
  - Updated pagination styling to utilize a grid layout, enhancing responsiveness and organization.
  - Adjusted input field styles for improved appearance and spacing.

- **Tests**
  - Expanded testing suite for the pagination component, including edge cases for input handling and pagination behavior.

### CODE REVIEW

File: src/components/Pagination/Pagination_shim.css [hunk 1]
```diff
@@ -32,9 +32,23 @@
...
 
 	.pagination__go-to-page {
-		display: flex;
+		display: grid;
+		grid-template-columns: 1fr auto auto;
 		align-items: center;
 		gap: 0.5rem;
+
+		.neo-input-editable__wrapper {
+			width: -webkit-fill-available;
```
Comment (yangAtSpoken) [inline src/components/Pagination/Pagination_shim.css:41]: Does not seem to matter.

**Review by yangAtSpoken — APPROVED** [review]
looks good to me

**Review by Justin-Avaya — APPROVED** [review]
MIA pagination component

**Comment by joe-s-avaya** [comment]
> MIA pagination component

Whoopsie, will fix, thanks!

**Comment by matthewAvaya** [comment]
Is it smart enough for the 'go to' page field to have a limit of characters a user can type? For example, if there are only 10 pages, the user can't type 100. 

Possibly out of scope:

- Arrows should be using chevron icons
- There should be 4px margin between the individual page index numbers
- Page count info (located on the left side) should read like this... 0 items selected / 240

**Comment by joe-s-avaya** [comment]
@matthewAvaya 

> Is it smart enough for the 'go to' page field to have a limit of characters a user can type? For example, if there are only 10 pages, the user can't type 100.

Yes. If you type in a number out of scope, it'll pull it down to the limit. Eg, if the limit is 1000 and you enter 2000, it'll show you page 1000.


> Possibly out of scope:

All out of scope for this PR.

**Comment by joe-s-avaya** [comment]
> MIA pagination component 

@Justin-Avaya fix has been pushed. Please check and confirm that it works as you would expect it to.

**Comment by Justin-Avaya** [comment]
> > MIA pagination component
> 
> @Justin-Avaya fix has been pushed. Please check and confirm that it works as you would expect it to.

- sort of fixed?



- the plus and minus buttons still behave oddly. expected the same behavior as letter characters.



- deleting the page number doesn't make a lot of sense to me.

File: src/components/Pagination/Nodes/GoToPage.tsx [hunk 2]
```diff
@@ -49,15 +49,14 @@ export const GoToPage = ({
 				}}
 				onKeyUp={(e) => {
 					if (e.key === Keys.ENTER) {
-						const v = Number.parseInt((e.target as HTMLInputElement).value, 10);
+						const v = (e.target as HTMLInputElement).valueAsNumber || 1;
 						onPageChange(e, v);
```
[the remark this answers is not in the document]
Comment (joe-s-avaya) [reply src/components/Pagination/Nodes/GoToPage.tsx]: updated, resolving

**Review by matthewAvaya — APPROVED** [review]

File: src/components/Pagination/Pagination_shim.css (hunk 1 above)
Comment (joe-s-avaya) [reply src/components/Pagination/Pagination_shim.css:41]: Huh, right you are, removed
